src/branchandbound.jl

```diff
@@ -0,0 +1,80 @@
+using TaylorModels
+# univariate
+function enclose_BranchandBound(f::Function, dom::Interval{T}; order::Int = 10,
+                                tol::Number = 0.6) where{T}
+    x0 = Interval(mid(dom))
+    x = TaylorModel1(order, x0, dom)
+    return branchandbound(f(x-x0).pol, dom - x0, tol)
+end
+# multivariate
+function enclose_BranchandBound(f::Function, dom::IntervalBox{T,N}; order::Int = 10,
+                                tol::Number = 0.6) where {T,N}
+    n = length(dom)
+    x0 = mid(dom)
+    set_variables(Float64, "x", order=2order, numvars=n)
+    x = [TaylorModelN(i, order, IntervalBox(x0), dom) for i=1:n]
+    return branchandbound(f(x...).pol, dom - x0, tol)
+end
+
+function branchandbound(p::Taylor1{T}, dom::Interval{W}, ϵ::Number) where{T,W}
+    return _branchandbound(p, dom, ϵ)
+end
+
+function branchandbound(p::TaylorN{T}, dom::IntervalBox{N,W}, ϵ::Number) where{T,N,W}
+    return _branchandbound(p, dom, ϵ)
+end
+
+function _branchandbound(p::Union{TaylorN{T},Taylor1{T}},
+                       dom::Union{IntervalBox{N,W},Interval{W}}, ϵ::Number) where {N,T,W}
+    K = 1
+    Rperv = evaluate(p, dom)
```

Comment:
what is particular of `p::Union{TaylorN{T},Taylor1{T}}` here? in other words, can you let `f::Function` and let (*) `evaluate(p, dom) -> enclose_TaylorModels(f, dom)`? 

(*) as a first step; later on we'd like to pass the options and more generally to use a custom solver

Reply:
I will try it locally first

Reply:
what does `Rperv` stand for? can you use more descriptive variable names?

Reply:
Bound over Range achieved by dividing domain on previous iteration

Reply:
ok so can you call it, for example, `Rprevious` or `Rprev`?

Reply:
Yes